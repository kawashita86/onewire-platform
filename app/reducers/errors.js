
export const ADD_ERROR = 'ADD_ERROR';
export const REMOVE_ERROR = 'REMOVE_ERROR';

export default function  errors(state = "", action) {
  switch (action.type) {

    case ADD_ERROR:
      return action.payload;

    case REMOVE_ERROR:
      return "";

    default:
      return state;
  }
}
