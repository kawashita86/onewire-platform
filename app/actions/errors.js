import {REMOVE_ERROR} from "../reducers/errors";


export function clearError() {
  return {
    type: REMOVE_ERROR
  };
}
