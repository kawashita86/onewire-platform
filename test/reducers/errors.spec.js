import errors from '../../app/reducers/errors';
import {
  ADD_ERROR,
  REMOVE_ERROR
} from '../../app/reducers/errors';

describe('reducers', () => {
  describe('errors', () => {
    it('should handle initial state', () => {
      expect(errors(undefined, {})).toMatchSnapshot();
    });

    it('should handle ADD_ERROR', () => {
      expect(errors(1, { type: ADD_ERROR })).toMatchSnapshot();
    });

    it('should handle REMOVE_ERROR', () => {
      expect(errors(1, { type: REMOVE_ERROR })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(errors(1, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
