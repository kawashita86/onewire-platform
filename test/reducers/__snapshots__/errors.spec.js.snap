// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers counter should handle initial state 1`] = `1`;

exports[`reducers counter should handle unknown action type 1`] = `1`;

exports[`reducers errors should handle ADD_ERROR 1`] = `1`;

exports[`reducers errors should handle REMOVE_ERROR 1`] = ``;

exports[`reducers errors should handle initial state 1`] = `""`;

exports[`reducers errors should handle unknown action type 1`] = `1`;
